File: Source/Utility/Hooks/use-sounds.hook.ts
import { SoundService, Workspace } from "@rbxts/services";

const PlayLocalSound = (Identity: number, Volume: number, Speed: number) => {
    const Sound = new Instance("Sound")

    Sound.SoundId = `rbxassetid://${Identity}`;
    Sound.PlaybackSpeed = Speed;
    Sound.Volume = Volume;

    Sound.Parent = Workspace;

    if (!Sound.IsLoaded) Sound.Loaded.Wait();

    SoundService.PlayLocalSound(Sound);

    task.delay(Sound.TimeLength, () => Sound.Destroy());
};

export function useSounds<Pack extends { [Key: string]: number }>(Sounds: Pack) {
    return (Name: keyof Pack, Volume: number = 0.5, Speed: number = 1) => {
        task.defer(PlayLocalSound, Sounds[Name], Volume, Speed);
    };
};

File: Output/Utility/Hooks/use-sounds.hook.lua
-- Compiled with roblox-ts v2.2.0
local TS = _G[script]
local _services = TS.import(script, TS.getModule(script, "@rbxts", "services"))
local SoundService = _services.SoundService
local Workspace = _services.Workspace
local PlayLocalSound = function(Identity, Volume, Speed)
	local Sound = Instance.new("Sound")
	Sound.SoundId = "rbxassetid://" .. tostring(Identity)
	Sound.PlaybackSpeed = Speed
	Sound.Volume = Volume
	Sound.Parent = Workspace
	if not Sound.IsLoaded then
		Sound.Loaded:Wait()
	end
	SoundService:PlayLocalSound(Sound)
	task.delay(Sound.TimeLength, function()
		return Sound:Destroy()
	end)
end
local function useSounds(Sounds)
	return function(Name, Volume, Speed)
		if Volume == nil then
			Volume = 0.5
		end
		if Speed == nil then
			Speed = 1
		end
		task.defer(PlayLocalSound, Sounds[Name], Volume, Speed)
	end
end
return {
	useSounds = useSounds,
}
